### PR TITLE
OUT 1017, OUT-1018 API Integration for Archive / Unarchive button

### PR DESCRIPTION
**Pull request by aatbip**
### Changes

- [x] Integrates APIs for the Archive and Unarchive toggle button
- [x] Adds last archived time field in the UI

### Testing Criteria

[Screencast from 18-11-24 01:25:43 PM +0545.webm](https://github.com/user-attachments/assets/1d169435-3033-4943-88db-3759ac65377c)

### CODE REVIEW

File: src/app/detail/[task_id]/[user_type]/page.tsx [hunk 1]
```diff
@@ -42,6 +42,8 @@ import { signedUrlTtl } from '@/types/constants'
 import { ActivityWrapper } from '@/app/detail/ui/ActivityWrapper'
 import { DeletedTaskRedirectPage } from '@/components/layouts/DeletedTaskRedirectPage'
 import { UserRole } from '@/app/api/core/types/user'
+import { ArchiveWrapper } from '../../ui/ArchiveWrapper'
+import { LastArchivedField } from '../../ui/LastArchiveField'
```
Comment (rrojan) [inline src/app/detail/[task_id]/[user_type]/page.tsx:46]: Use relative imports

File: src/app/detail/ui/ArchiveWrapper.tsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+          })
+
+          // Re-fetch updated data
+          return await fetcher(cacheKey)
+        },
+        {
+          optimisticData: { task: optimisticTask }, // Update UI immediately
+          rollbackOnError: true, // Revert on error
+          revalidate: true, // Ensure data consistency
+        },
```
Comment (rrojan) [inline src/app/detail/ui/ArchiveWrapper.tsx:54]: How are we checking for realtime archive updates?

File: src/components/buttons/ArchiveBtn.tsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  return (
+    <Stack
+      direction="row"
+      alignItems="center"
+      columnGap="6px"
+      sx={{
+        padding: '3px 8px',
+        ':hover': {
+          cursor: 'pointer',
+          border: ' 1px solid var(--Black-Alpha-1, rgba(0, 0, 0, 0.12))',
```
Comment (rrojan) [inline src/components/buttons/ArchiveBtn.tsx:16]: > var(--Black-Alpha-1

Is this defined in our code? I don't see global CSS variables... I think we should be using theme here?

**Review by rrojan — CHANGES_REQUESTED** [review]

File: src/app/detail/ui/ArchiveWrapper.tsx [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+  const { data, error } = useSWR(cacheKey, fetcher)
+  const [isArchived, setIsArchived] = useState<boolean | undefined>(undefined)
+
+  // Set the initial state when `data` becomes available
+  useEffect(() => {
+    if (data?.task) {
+      setIsArchived(data.task.isArchived)
+      store.dispatch(setTask(data.task))
+    }
+  }, [data])
```
Comment (rrojan) [inline src/app/detail/ui/ArchiveWrapper.tsx:26]: I don't think this is listening for realtime archive status updates

File: src/app/detail/ui/ArchiveWrapper.tsx (hunk 2 above)
Comment (aatbip) [reply src/app/detail/ui/ArchiveWrapper.tsx]: Removed `revalidate:true` in the commit below

File: src/app/detail/ui/ArchiveWrapper.tsx (hunk 4 above)
Comment (aatbip) [reply src/app/detail/ui/ArchiveWrapper.tsx]: Added real time in the commit below

File: src/app/detail/ui/ArchiveWrapper.tsx [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    }
+  }, [tasks, taskId])
+
+  const handleToggleArchive = async () => {
+    if (isArchived === undefined) return // Prevent toggling if state isn't initialized yet
+
+    const newIsArchived = !isArchived
+    const optimisticTask = { ...task, isArchived: newIsArchived }
+
+    setIsArchived(newIsArchived) // Optimistically update local state
```
Comment (rrojan) [inline src/app/detail/ui/ArchiveWrapper.tsx:35]: Is there a reason we aren't using `task` directly for `isArchived`?

**Review by rrojan — CHANGES_REQUESTED** [review]

**Review by rrojan — CHANGES_REQUESTED** [review]
The time ago text does not change until refresh. I thought we agreed to implement a 60-second change for time ago text as per the task description?

image

**Review by rrojan — APPROVED** [review]
LGTM :trophy:

**Pull request merged**